src/main/java/org/cactoos/list/ImmutableListIterator.java

```diff
@@ -23,88 +23,59 @@
  */
 package org.cactoos.list;
 
-import java.util.List;
 import java.util.ListIterator;
-import org.cactoos.Scalar;
-import org.cactoos.scalar.Sticky;
-import org.cactoos.scalar.Unchecked;
 
 /**
- * List Iterator that doesn't allow mutations.
+ * Immutable {@link ListIterator} that doesn't allow mutations.
  *
  * <p>There is no thread-safety guarantee.
  *
  * @param <T> Items type
- * @since 0.35
+ * @since 1.0
  */
 public final class ImmutableListIterator<T> implements ListIterator<T> {
 
     /**
      * Original list iterator.
      */
-    private final Unchecked<ListIterator<T>> origin;
-
-    /**
-     * Ctor.
-     * @param list List that will be called to get a list iterator.
-     */
-    public ImmutableListIterator(final List<T> list) {
-        this(list::listIterator);
-    }
-
-    /**
-     * Ctor.
-     * @param list List that will be called to get a list iterator.
-     * @param index Start index for a newly created list iterator.
-     */
-    public ImmutableListIterator(final List<T> list, final int index) {
-        this(() -> list.listIterator(index));
-    }
+    private final ListIterator<T> origin;
 
     /**
      * Ctor.
      * @param iter Original list iterator.
      */
     public ImmutableListIterator(final ListIterator<T> iter) {
-        this(() -> iter);
-    }
-
-    /**
-     * Ctor.
-     * @param orig Original list iterator.
-     */
-    public ImmutableListIterator(final Scalar<ListIterator<T>> orig) {
-        this.origin = new Unchecked<>(new Sticky<>(orig));
+        this.origin = new ListIteratorOf<>(iter);
```

Comment:
@fanifieiev why do you wrap it in this? It's not needed, is it? It's better to implement here directly the immutability instead of relying on a class that is going to change…